### PR TITLE
Added gcp plugins

### PR DESCRIPTION
**Pull request by evalsocket**
- Added GCP BigQuery plugins documentation
- Rename task template to web API in plugin docs 

Fix: https://github.com/flyteorg/flyte/issues/1771

Dependent : https://github.com/flyteorg/flyte/issues/1776

### CODE REVIEW

File: rsts/deployment/plugin_setup/gcp/bigquery.rst [hunk 1]
```diff
@@ -0,0 +1,50 @@
+.. _deployment-plugin-setup-gcp-bigquery:
+
+Google Bigquery Plugin Setup
+----------------------------
+
+This guide gives an overview of how to set up Big query in your Flyte deployment. Big query plugin needs Flyte deployment in GCP cloud; sandbox/GCP/Azure wouldn't work.
```
Comment (samhita-alla) [inline rsts/deployment/plugin_setup/gcp/bigquery.rst:6]:
```suggestion
This guide gives an overview of how to set up Big query in your Flyte deployment. Big query plugin needs Flyte deployment in GCP cloud; sandbox/AWS/Azure wouldn't work.
```

File: rsts/deployment/plugin_setup/gcp/bigquery.rst [hunk 2]
```diff
@@ -0,0 +1,50 @@
+.. _deployment-plugin-setup-gcp-bigquery:
+
+Google Bigquery Plugin Setup
```
Comment (samhita-alla) [inline rsts/deployment/plugin_setup/gcp/bigquery.rst:3]:
```suggestion
Google BigQuery Plugin Setup
```

File: rsts/deployment/plugin_setup/gcp/bigquery.rst [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+              container_array: k8s-array
+              bigquery: bigquery
+
+2. Upgrade the Flyte Helm release.
+
+   .. code-block:: bash
+
+      helm upgrade -n flyte -f values-bigquery.yaml flyteorg/flyte
+
+3. Register the Big query plugin example.
```
Comment (samhita-alla) [inline rsts/deployment/plugin_setup/gcp/bigquery.rst:37]:
```suggestion
3. Register the BigQuery plugin example.
```

File: rsts/deployment/plugin_setup/gcp/index.rst [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+#################
+
+
+.. panels::
+    :header: text-center
+
+
+    .. link-button:: deployment-plugin-setup-gcp-bigquery
+       :type: ref
+       :text: Google Bigquery
```
Comment (samhita-alla) [inline rsts/deployment/plugin_setup/gcp/index.rst:14]:
```suggestion
       :text: Google BigQuery
```

**Review by samhita-alla — APPROVED** [review]

**Comment by evalsocket** [comment]
@samhita-alla Added a small note. Need +1

**Review by samhita-alla — APPROVED** [review]

**Pull request merged**